test/system_benchmarks.cc

```diff
@@ -139,3 +142,126 @@ TEMPLATE_TEST_CASE_METHOD_SIG(
+		buffer<float, 2> rad(rad_data.data(), rad_size);
+		// set the first line of rad to 1
+		queue.submit([&](handler& cgh) {
+			auto write_rad_mapper = [](chunk<1> chnk) -> subrange<2> {
+				return {
+				    id(0, chnk.offset.get(0)),  // offset
+				    range(1, chnk.range.get(0)) // range
+				};
+			};
+			accessor write_rad{rad, cgh, write_rad_mapper, write_only, no_init};
+			cgh.parallel_for(n_tris, [=](item<1> item) { write_rad[{0, item.get_id(0)}] = 1.f; });
+		});
+
+		// main simulation loop
+		for(size_t t = 1; t < num_iter; ++t) {
+			queue.submit([&](handler& cgh) {
+				// read everything written before the current timestep
+				auto read_rad_mapper = [t](chunk<2> chnk) -> subrange<2> {
+					return {
+					    id(0, 0),                         // offset
+					    range(t, chnk.global_size.get(0)) // range
+					};
+				};
+				// only need to write to radiosities of own triangles in current timestep
+				auto write_rad_mapper = [t](chunk<2> chnk) -> subrange<2> {
+					return {
+					    id(t, chnk.offset.get(0)),  // offset
+					    range(1, chnk.range.get(0)) // range
+					};
+				};
+
+				accessor read_kij{kij, cgh, access::one_to_one(), read_only};
+				accessor write_rad{rad, cgh, write_rad_mapper, write_only, no_init};
+				accessor read_rad{rad, cgh, read_rad_mapper, read_only};
+
+				cgh.parallel_for(kij_size, [=](item<2> item) {
+					float val = 0.f;
+					float included_items = 0.f;
+					for(size_t i = 0; i < t; ++i) {
+						// printf("i: %lu, t: %lu, id: %lu/%lu, read_rad: %f, read_kij: %f\n", i, t, item.get_id(0), item.get_id(1), read_rad[{i,
+						// item.get_id(0)}], read_kij[{item.get_id(0), item.get_id(1)}]);
+						val += read_rad[{i, item.get_id(0)}] * read_kij[{item.get_id(0), item.get_id(1)}];
+					}
+					val /= (float)t;
+					write_rad[{t, item.get_id(0)}] = val;
+				});
+			});
+		}
+
+		if(perform_verification_copy) {
+			queue.submit([&](handler& cgh) {
+				accessor read_rad{rad, cgh, access::one_to_one(), read_only};
+				accessor write_rad_result{rad_result, cgh, access::one_to_one(), write_only};
+				cgh.parallel_for(rad_size, [=](item<2> item) { write_rad_result[item] = read_rad[item]; });
+			});
+		}
+		queue.wait();
+	};
+
+	benchmark_run(true);
+
+	BENCHMARK("iterations") { benchmark_run(false); };
+
+	// check result
+	buffer<bool, 0> success_buffer = true;
+	queue.submit([&](handler& cgh) {
+		accessor r{rad_result, cgh, access::all{}, read_only_host_task};
+		accessor succ{success_buffer, cgh, access::all{}, write_only_host_task};
+		cgh.host_task(on_master_node, [=] {
+			experimental::for_each_item(rad_size, [=](item<2> item) {
+				const float expected = 1.f;
+				constexpr float epsilon = 0.01f;
+				if(std::fabs(r[item] - expected) > epsilon) {
+					fmt::print("Mismatch at {}/{}: {} !~= {} +/- {}\n", item.get_id(0), item.get_id(1), r[item], expected, epsilon);
+					succ = false;
+				}
+			});
+		});
+	});
+	CHECK(*queue.fence(success_buffer).get() == true);
```

Comment:
Could be a lot simpler if we just fence'd `rad_result` and did the comparison in the main thread.

Reply:
I'm now doing the verification in the lambda as per the original idea of @psalz .